### PR TITLE
17.0 itfixdownpayment jco

### PR DESCRIPTION
**Pull request by jco-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by gawa-odoo — COMMENTED** [review]
Tell me what you think about it.
I'm not really sure it's best.

File: addons/l10n_it_edi/models/account_move.py [hunk 1]
```diff
@@ -243,9 +243,15 @@ def _l10n_it_edi_get_line_values(self, reverse_charge_refund=False, is_downpayme
         """
         invoice_lines = []
         lines = self.invoice_line_ids.filtered(lambda l: l.display_type not in ('line_note', 'line_section'))
+        downpayment_lines = self.env['account.move.line']
+        if not is_downpayment:
```
[the remark this answers is not in the document]
Comment (jco-odoo) [reply addons/l10n_it_edi/models/account_move.py:247]: I was thinking the lines to be the easier to filter out than the baselinedicts, but maybe not.

**Comment by jco-odoo** [comment]
@gawa-odoo Updated

File: addons/l10n_it_edi/models/account_move.py [hunk 2]
```diff
@@ -262,9 +262,18 @@ def _l10n_it_edi_get_line_values(self, reverse_charge_refund=False, is_downpayme
         template = self.env['ir.qweb']._load('l10n_it_edi.account_invoice_line_it_FatturaPA')[0]
         flat_discount_element = template.find('.//ScontoMaggiorazione/Percentuale')
         if flat_discount_element is not None:
+            downpayment_lines = []
+            if not is_downpayment:
+                # Negative lines linked to down payment should stay negative
+                for line_dict in base_lines:
+                    line = line_dict['record']
+                    if line.price_subtotal < 0 or line._get_downpayment_lines():
```
Comment (gawa-odoo) [inline addons/l10n_it_edi/models/account_move.py:270]: I think it's
```suggestion
                    if line.price_subtotal < 0 and line._get_downpayment_lines():
```

**Review by gawa-odoo — COMMENTED** [review]
Just a thing

File: addons/l10n_it_edi/models/account_move.py (hunk 2 above)
Comment (jco-odoo) [reply addons/l10n_it_edi/models/account_move.py]: yes, thanks for noticing

**Review by gawa-odoo — COMMENTED** [review]
@robodoo r+

In the future, we could try to see if this part https://github.com/odoo/odoo/pull/186755/files#diff-7a314f6fb9514fd24adb9e88e00fe802ff75743c7227d648a65f44ca34de9883R291-R297 could be included in what you added, but that's unrelated to this fix (we would need to add a new key in the dict etc... )

**Pull request closed without merging**